coreneuron/nrniv/main1.cpp

```diff
@@ -368,15 +372,27 @@ const char* nrn_version(int) {
 
 using namespace coreneuron;
 
-extern "C" void mk_mech_init(int argc, char** argv) {
-#if NRNMPI
-    nrnmpi_init(1, &argc, &argv);
-#endif
+
+extern "C" int mk_mech_init(int argc, char** argv) {
+    #if NRNMPI
+        nrnmpi_init(1, &argc, &argv);
+    #endif
     // read command line parameters and parameter config files
-    nrnopt_parse(argc, (const char**)argv);
+
+    cn_par.parse(argc, argv);
+
+    if (cn_par.print_arg) {
```

Comment:
I was wondering if the parameters should be always printed, since this is what was happening before. I also got some issues with printing the parameters two times and some of them being printed wrong when running for example `tests/integration/ring/integration_test.sh` with `--show`. Do you have the same problem?

Reply:
I kept the same settings as the previous implementation, so they get printed every time as a confirmation of what is happening. I'll check what happen in the case you described.